File: ruby/iq10/iq10.rb
#!/usr/bin/ruby
$letter = {"1" => '1', "2" => "ABC", "3" => "DEF", "4" => "GHI", "5" => "JKL", "6" => "MNO", "7" => "PRS", "8" => "TUV", "9" => "WXY", "0" => "0"}
def testmethod(cur_digit, chars = '')
	if cur_digit == $len - 1
		letters = $letter[$number[cur_digit]]
		letters.split("").each do |item|
			$file.write(chars + item + "\n")
		end
	else
		letters = $letter[$number[cur_digit]]
		letters.split("").each do |item|
			testmethod(cur_digit + 1, chars + item)
		end		
	end	
end

def main()
	while true
		puts("Enter your 7 digit telephone number")
		$number = gets().chomp
		$number.gsub!("-","")
		if $number.length == 7
			$len = 7
    		break 
		end
	end
	$file = File.open("word", "w")
	testmethod(0)
	$file.close	 
end
main()

File: python/iq10/iq10.py
#!/usr/bin/python
FINAL_NUM_LEN = 7
letter = {'1': '1', '2':'ABC', '3':'DEF', '4':'GHI', '5':'JKL', '6':'MNO', '7':'PRS', '8':'TUV', '9':'WXY'}
fp = open("words", "w")
def num_toStr(cur_digit, chars = ''):
  global FINAL_NUM_LEN
  global letter
  global number
  global fp
  if cur_digit == FINAL_NUM_LEN - 1:
    letters = letter[number[cur_digit]]
    for i in letters:
      fp.write(chars + i + "\n")
  else:
    letters = letter[number[cur_digit]]
    for i in letters:
      num_toStr(cur_digit + 1, chars + i)


def main():
  global number
  global fp
  while(True):
    number  = raw_input("Enter your 7 digit phone number").replace("-","")
    if len(number) == 7:    
      break
  num_toStr(0)
  fp.close()

if __name__ == "__main__":
  main()
    		
